modules/web/src/com/haulmont/cuba/web/sys/navigation/EditorTypeExtractor.java

```diff
@@ -17,111 +17,63 @@
 package com.haulmont.cuba.web.sys.navigation;
 
 import com.haulmont.cuba.core.entity.Entity;
-import com.haulmont.cuba.gui.components.AbstractEditor;
 import com.haulmont.cuba.gui.config.WindowInfo;
 import com.haulmont.cuba.gui.screen.EditorScreen;
-import com.haulmont.cuba.gui.screen.StandardEditor;
+import org.springframework.core.ResolvableType;
 
 import javax.annotation.Nullable;
-import java.lang.reflect.ParameterizedType;
-import java.lang.reflect.Type;
-import java.util.Arrays;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
 
 public final class EditorTypeExtractor {
 
+    private static final Map<Class<?>, Integer> EDITOR_CLASSES = new HashMap<>();
```

Comment:
I don't quite get why do we need this map. I may assume it's to make traversing code more generic, IMO it's not necessary.